src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown

```diff
@@ -0,0 +1,373 @@
+  - audio
+  - codec
+  - codecs
+  - media source extensions
+  - MSE
+permalink: /2015/06/mse-gapless-audio
+---
+
+# Media Source Extensions for Audio: Eliminating the Gap
+
+## Introduction
+
+[Media Source Extensions (MSE)](http://dvcs.w3.org/hg/html-media/raw-file/tip/media-source/media-source.html) provide extended buffering and playback control for the HTML5 <audio> and <video> elements. While originally developed to facilitate [Dynamic Adaptive Streaming over HTTP (DASH)](http://dashif.org/mpeg-dash/) based video players, below we'll see how they can be used for audio; specifically for [gapless playback](http://en.wikipedia.org/wiki/Gapless_playback).
+
+You've likely listened to a music album where songs flowed seamlessly across tracks; you may even be listening to one right now. Artists create these [gapless playback](http://en.wikipedia.org/wiki/Gapless_playback) experiences both as an artistic choice as well as an artifact of [vinyl records](http://en.wikipedia.org/wiki/Gramophone_record) and [CDs](http://en.wikipedia.org/wiki/Compact_disc) where audio was written as one continuous stream. Unfortunately, due to the way modern audio codecs like [MP3](http://en.wikipedia.org/wiki/MP3) and [AAC](http://en.wikipedia.org/wiki/Advanced_Audio_Coding) work, this seamless aural experience is often lost today.
+
+We'll get into the details of why below, but for now lets start with a demonstration. Below is the first thirty seconds of the excellent [Sintel](http://www.sintel.org/) chopped into five separate MP3 files and reassembled using MSE. The red lines indicate gaps introduced during the creation (encoding) of each MP3; you'll hear glitches at these points.
+
+<iframe seamless style="border: 0; height: 335px; width: 100%" src="https://simpl.info/mse/audio/gap"></iframe>
+
+Yuck! That's not a great experience; we can do better. With a little more work, using the exact same MP3 files in the above demo, we can use MSE to remove those annoying gaps. The green lines in the next demo indicate where the files have been joined and the gaps removed. On Chrome 38+ this will playback seamlessly!
+
+<iframe seamless style="border: 0; height: 335px; width: 100%" src="https://simpl.info/mse/audio/gapless"></iframe>
+
+There are a [variety of ways to create gapless content](#creating-gapless-content). For the purposes of this demo, we'll focus on the type of files a normal user might have lying around. Where each file has been encoded separately without regard for the audio segments before or after it.
+
+## Basic Setup
+
+First, lets backtrack and cover the basic setup of a MediaSource instance. Media Source Extensions, as the name implies, are just extensions to the existing media elements. Below, we're assigning an [Object URL](https://developer.mozilla.org/en-US/docs/Web/API/URL.createObjectURL), representing our MediaSource instance, to the source attribute of an audio element; just like you would set a standard URL.
+
+{% highlight javascript %}
```

Comment:
Looks like the content Dale has used is somewhat different from the source code.